File: frontend/src/api/workflows.ts
import { apiPost } from './http'

export type Priority = 'low' | 'medium' | 'high' | 'urgent'

export type TextWorkflowRequest = {
  content: string
}

export type TaskDraft = {
  title: string
  assignee?: string | null
  due_date?: string | null
  priority: Priority
  description: string
}

export type TicketDraft = {
  ticket_type: string
  title: string
  description: string
  priority: Priority
  suggested_action?: string | null
}

export type QualityIssueTicketDraft = {
  ticket_type: string
  title: string
  description: string
  priority: Priority
  abnormality_type: string
  affected_scope: string
  initial_disposition: string
  required_actions: string[]
}

export type PurchaseDraft = {
  item_name?: string | null
  quantity?: number | null
  reason?: string | null
  budget?: number | null
  supplier?: string | null
}

export type WeeklyReportDraft = {
  progress: string
  issues: string
  quality_risks: string
  equipment_risks: string
  purchase_risks: string
  next_plan: string
  decisions_needed: string
}

export type MeetingToTasksResponse = {
  source: string
  tasks: TaskDraft[]
  workflow_status: string
  requires_approval: boolean
  message: string
}

export type MaintenanceTicketResponse = {
  source: string
  ticket_draft?: TicketDraft | null
  workflow_status: string
  requires_approval: boolean
  message: string
}

export type QualityIssueResponse = {
  source: string
  quality_issue_draft?: QualityIssueTicketDraft | null
  workflow_status: string
  requires_approval: boolean
  message: string
}

export type PurchaseWorkflowResponse = {
  source: string
  purchase_draft?: PurchaseDraft | null
  missing_fields: string[]
  workflow_status: string
  requires_approval: boolean
  message: string
}

export type WeeklyReportResponse = {
  source: string
  weekly_report?: WeeklyReportDraft | null
  message: string
}

export function generateMeetingTasks(content: string) {
  return apiPost<MeetingToTasksResponse, TextWorkflowRequest>('/workflows/meeting-to-tasks', { content })
}

export function generateMaintenanceTicket(content: string) {
  return apiPost<MaintenanceTicketResponse, TextWorkflowRequest>('/workflows/maintenance-ticket', { content })
}

export function generateQualityIssue(content: string) {
  return apiPost<QualityIssueResponse, TextWorkflowRequest>('/workflows/quality-issue', { content })
}

export function generatePurchaseRequest(content: string) {
  return apiPost<PurchaseWorkflowResponse, TextWorkflowRequest>('/workflows/purchase-request', { content })
}

export function generateWeeklyReport(content: string) {
  return apiPost<WeeklyReportResponse, TextWorkflowRequest>('/workflows/weekly-report', { content })
}

File: backend/app/api/v1/endpoints/workflows.py
from fastapi import APIRouter

from app.schemas.workflow import (
    MaintenanceTicketResponse,
    MeetingToTasksResponse,
    PurchaseWorkflowResponse,
    QualityIssueResponse,
    TextWorkflowRequest,
    WeeklyReportResponse,
)
from app.workflows.maintenance_ticket import generate_maintenance_ticket
from app.workflows.meeting_to_tasks import generate_task_drafts
from app.workflows.purchase_request import generate_purchase_request
from app.workflows.quality_issue import generate_quality_issue_ticket
from app.workflows.weekly_report import generate_weekly_report

router = APIRouter()


@router.post("/meeting-to-tasks", response_model=MeetingToTasksResponse, summary="会议纪要转任务")
def meeting_to_tasks(request: TextWorkflowRequest) -> MeetingToTasksResponse:
    """从会议纪要中提取任务草稿，后续进入人工确认。"""
    return generate_task_drafts(request.content)


@router.post("/maintenance-ticket", response_model=MaintenanceTicketResponse, summary="设备异常转维修工单")
def maintenance_ticket(request: TextWorkflowRequest) -> MaintenanceTicketResponse:
    """根据设备异常描述生成维修工单草稿。"""
    return generate_maintenance_ticket(request.content)


@router.post("/quality-issue", response_model=QualityIssueResponse, summary="质量异常转处理工单")
def quality_issue(request: TextWorkflowRequest) -> QualityIssueResponse:
    """根据质量异常描述生成 NCR/质量异常工单草稿。"""
    return generate_quality_issue_ticket(request.content)


@router.post("/purchase-request", response_model=PurchaseWorkflowResponse, summary="采购需求转申请草稿")
def purchase_request(request: TextWorkflowRequest) -> PurchaseWorkflowResponse:
    """根据采购需求生成采购申请草稿，必要时提示补充字段。"""
    return generate_purchase_request(request.content)


@router.post("/weekly-report", response_model=WeeklyReportResponse, summary="生成项目周报")
def weekly_report(request: TextWorkflowRequest) -> WeeklyReportResponse:
    """根据项目记录生成周报草稿。"""
    return generate_weekly_report(request.content)
